frontend/src/components/query/query-builder/steps/order/order-step.tsx

```diff
@@ -0,0 +1,360 @@
+"use client";
+import {
+  OrderColumn,
+  OrderStep,
+  InferSchemaOutputSuccess,
+  InferredSchemaColumn,
+  Pipeline,
+  Step,
+  StepIdentifierEnum,
+} from "@/definitions";
+import {
+  Button,
+  Menu,
+  MenuItem,
+  Popover,
+  Section,
+  Text,
+} from "@blueprintjs/core";
+import { ItemRenderer, Select } from "@blueprintjs/select";
+import Loading from "@/app/loading";
+import { ErrorDisplay } from "@/components/error-display";
+import { NewStepSelection } from "../../query-builder";
+import InvalidStepPopover from "../../invalid-step-popover";
+import OrderCaseTag from "./order-case-tag";
+import SingleColumnSelector from "@/components/column-selectors/single-column-selector/single-column-selector";
+import { usePipelineSchema } from "@/data/use-user-query";
+import { useState, useEffect } from "react";
+import * as _ from "lodash";
+
+interface OrderStepProps {
+  index: number;
+  step: OrderStep | null;
+  pipeline: Pipeline;
+  setPipeline: (value: Pipeline) => void;
+  edit: boolean;
+  setEditStepIndex: (value: number | null) => void;
+  setNewStepType: (value: NewStepSelection | null) => void;
+  create?: boolean;
+}
+
+export default function OrderStepComponent({
+  index,
+  step,
+  pipeline,
+  setPipeline,
+  edit,
+  setEditStepIndex,
+  setNewStepType,
+  create,
+}: OrderStepProps) {
+  const [orderCases, setOrderCases] = useState<OrderColumn[]>([]);
+  const [selectedColumn, setSelectedColumn] =
+    useState<InferredSchemaColumn | null>(null);
+  const [selectedDirection, setSelectedDirection] = useState<
```

Comment:
Can't we default to a direction? Prob should default asc

Reply:
Sounds good to me